### ISSUE TITLE
Core/SAI: It's necessary to implement TARGET_INVOKER_ SUMMONED as the target in SAI

### ISSUE THREAD

**Issue by carl3175** [TrinityCore/TrinityCore#30307]
### Description

In many quests, it's necessary to use the SMARTAI_TARGETS with TARGET_INVOKER_ SUMMONED as the target in SAI. For example, pets or guardians summoned by characters, and so on. Can we prioritize implementing this feature? Thanks a lot ~

### Expected behaviour

TARGET_INVOKER_ SUMMONED implemented

### Steps to reproduce the problem

try SAI

### Branch

master

### TC rev. hash/commit

TrinityCore rev. 7705e5eeee70 2024-10-02 01:05:35 +0200 (master branch) (Windows, AMD64, Release, Static) (worldserver-daemon)

### Operating system

Win11 23H2

### Custom changes

None

**Comment by jackpoz** [comment]
Could you please include a small SAI script to test the implementation for those devs who might want to implement it ?

**Comment by carl3175** [comment]
> Could you please include a small SAI script to test the implementation for those devs who might want to implement it ?

Of course, in quest (26028) 'Mr. D's Wild Ride', The character will summon Clarissa to follow, and when quest rewarded, the summoned Clarissa(41984) should Despawn.Here are some SAI script：

`INSERT INTO `world`.`smart_scripts` (`entryorguid`, `source_type`, `id`, `link`, `Difficulties`, `event_type`, `event_phase_mask`, `event_chance`, `event_flags`, `event_param1`, `event_param2`, `event_param3`, `event_param4`, `event_param5`, `event_param_string`, `action_type`, `action_param1`, `action_param2`, `action_param3`, `action_param4`, `action_param5`, `action_param6`, `action_param7`, `action_param_string`, `target_type`, `target_param1`, `target_param2`, `target_param3`, `target_param4`, `target_param_string`, `target_x`, `target_y`, `target_z`, `target_o`, `comment`) VALUES 
(41023, 0, 6, 0, '', 20, 0, 100, 0, 26028, 0, 0, 0, 0, '', 41, 0, 0, 0, 0, 0, 0, 0, NULL, 19, 41984, 50, 0, 0, NULL, 0, 0, 0, 0, 'Overlord Krom\'gar - On Quest Reward - Despawn (Clarissa)'),`

It should be noted that this type of SMARTAI-TARGETS is not accurate enough in locating the target, and there may be errors if multiple characters nearby are doingthis quest at the same time.
